### ISSUE TITLE
NHibernateStorageSession assumes saga connectionstring

### ISSUE THREAD

**Issue by dvdstelt** [Particular/NServiceBus.NHibernate#286]
Together @HEskandari , I investigated an issue related to [this StackOverflow issue](http://stackoverflow.com/questions/43680632/why-do-i-need-any-saga-storage) and came to the following conclusion:

### Issue description
When you try to disable sagas and subscriptions, are left with just timeouts and provide a connectionstring for just timeouts, the persister complains about saga connectionstring not being found.

### Detailed description
The [NHibernatePersistence](https://github.com/Particular/NServiceBus.NHibernate/blob/master/src/NServiceBus.NHibernate/NHibernatePersistence.cs) feature tries to enable timeouts, sagas, subscriptions, etc. by default. However line 21 enables `NHibernateStorageSession` and this is looking for a specific saga connectionstring in [line 35](https://github.com/Particular/NServiceBus.NHibernate/blob/e21340aca078790d76a1d3d90fc3e0596b43ca02/src/NServiceBus.NHibernate/SynchronizedStorage/NHibernateStorageSession.cs#L35).

When just providing the connectionstring for the timeout, after disabling everything else, it complains about a missing saga connectionstring.

```
        endpointConfiguration.DisableFeature<MessageDrivenSubscriptions>();
        endpointConfiguration.DisableFeature<Sagas>();
        endpointConfiguration.UsePersistence<NHibernatePersistence, StorageType.Timeouts>();
```
and configuration
```
<add name="NServiceBus/Persistence/NHibernate/Timeout" connectionString="server=.\sqlexpress;database=nservicebus; Trusted_Connection=True;" />
```

### Workaround

Disabling the feature `NHibernateStorageSession` like this
```
        endpointConfiguration.DisableFeature<NHibernateStorageSession>();
        endpointConfiguration.DisableFeature<MessageDrivenSubscriptions>();
        endpointConfiguration.DisableFeature<Sagas>();
        endpointConfiguration.UsePersistence<NHibernatePersistence, StorageType.Timeouts>();
```

### Suggested solution

Easiest solution is probably to make sure `NHibernateStorageSession` doesn't require the saga connectionstring. Maybe just have "Saga" empty in line 35. That would also require to fix the error message, since it might not have a suffix at all.

**Comment by tmasternak** [comment]
I don't think we should tackle it right now. We have a workaround and a single client that had issues with it.
